c2rust-analyze/src/rewrite/expr/distribute.rs

```diff
@@ -92,19 +105,27 @@ pub fn distribute<'tcx>(
                     loc,
                     sub_loc: key.1,
                     desc: origin.desc,
+                    priority,
                 });
         }
     }
 
-    // If a single `HirId` has rewrites from multiple different pieces of MIR, it's ambiguous how
-    // to order those rewrites.  (`mir_rewrites` only establishes an ordering between rewrites on
-    // the same `Location`.)  For now, we complain if we see this ambiguity; in the future, we may
-    // need to add rules to resolve it in a particular way, such as prioritizing one `SubLoc` or
-    // `MirOriginDesc` over another.
-    for (&hir_id, infos) in &info_map {
-        let all_same_loc = infos.iter().skip(1).all(|i| {
-            i.loc == infos[0].loc && i.sub_loc == infos[0].sub_loc && i.desc == infos[0].desc
-        });
+    // If a single `HirId` has rewrites from multiple different pieces of MIR at the same
+    // `Priority`, it's ambiguous how to order those rewrites.  (`mir_rewrites` only establishes an
+    // ordering between rewrites on the same `Location`.)  For now, we complain if we see this
+    // ambiguity; in the future, we may need to add rules to resolve it in a particular way, such
+    // as prioritizing one `SubLoc` over another.
+    for (&hir_id, infos) in &mut info_map {
+        infos.sort_by_key(|i| i.priority);
+        let mut all_same_loc = true;
+        for (a, b) in infos.iter().zip(infos.iter().skip(1)) {
```

Comment:
```suggestion
        for [a, b] in infos.windows(2).map(|w| [&w[0], &w[1]]) {
```

I think the intent with this would be clearer vs. the zipping way.

Reply:
I guess that works - I'm just used to the Haskell version `zip xs (tail xs)`.  It's a shame `<[T]>::array_windows` isn't stable yet (we could avoid the `map`), but this version is fine.

Reply:
Yeah, I tried to use `.array_windows`, too, but almost all of those array versions are still unstable unfortunately.